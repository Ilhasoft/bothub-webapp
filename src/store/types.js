export default {
  SET_TOKEN: 'SET_TOKEN',
  OPEN_LOGIN_MODAL: 'OPEN_LOGIN_MODAL',
  CLOSE_LOGIN_MODAL: 'CLOSE_LOGIN_MODAL',
  SET_PROFILE: 'SET_PROFILE',
  SET_NICKNAME_AUTHENTICATED: 'SET_NICKNAME_AUTHENTICATED',
  SET_REPOSITORY_RELATED_UUID: 'SET_REPOSITORY_RELATED_UUID',
  CACHE_FETCH: 'CACHE_FETCH',
};
