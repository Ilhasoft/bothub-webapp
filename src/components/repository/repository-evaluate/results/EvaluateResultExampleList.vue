<template>
  <div>
    <h3
      class="evaluate-result-example-list__title">
      {{ $t('webapp.result.sentence_details') }}
    </h3>
    <div
      v-if="busy"
      class="evaluate-result-example-list__loading">
      <loading/>
    </div>
    <p v-else-if="error">{{ $t('webapp.result.error') }}</p>
    <div v-else-if="resultExampleList && resultExampleList.length > 0">
      <p>
        {{ $t('webapp.result.sentence_details_text') }}
      </p>
      <evaluate-result-example-item
        v-for="(item, i) in resultExampleList"
        id="tour-evaluate_result-step-0"
        :key="i"
        :text="item.text"
        :intent="item.intent"
        :confidence="item.intent_prediction.confidence"
        :intent-success="(item.intent_status || item.status) === 'success'"
        :success="isSuccess(item)"
        :intent-prediction="item.intent_prediction"
        :entities="item.true_entities || []"
        :added-entities="item.false_positive_entities || []"
        :swapped-entities="item.swapped_error_entities || []"/>
      <div class="evaluate-result-example-list__pagination">
        <b-pagination
          v-if="resultExampleList.length > 0"
          :range-before="4"
          :range-after="4"
          :total="total"
          :current.sync="page"
          :per-page="limit"
          aria-next-label="Next page"
          aria-previous-label="Previous page"
          aria-page-label="Page"
          aria-current-label="Current page"/>
      </div>
    </div>
    <p v-else>{{ $t('webapp.result.do_not_log') }}</p>
    <tour
      v-if="activeTutorial === 'evaluate' && resultExampleList.length > 0"
      :step-count="1"
      name="evaluate_result"/>
  </div>
</template>

<script>
import { mapActions, mapState, mapGetters } from 'vuex';
import EvaluateResultExampleItem from '@/components/repository/repository-evaluate/results/EvaluateResultExampleItem';
import Loading from '@/components/shared/Loading';
import Tour from '@/components/Tour';

export default {
  name: 'EvaluateResultExampleList',
  components: {
    EvaluateResultExampleItem,
    Loading,
    Tour,
  },
  props: {
    id: {
      type: Number,
      required: true,
    },
  },
  data() {
    return {
      resultExampleList: [],
      limit: 10,
      busy: false,
      error: null,
      pages: 0,
      page: 1,
    };
  },
  computed: {
    ...mapState({
      repository: state => state.Repository.selectedRepository,
    }),
    ...mapGetters([
      'activeTutorial',
    ]),
    total() {
      return this.limit * this.pages;
    },
  },
  watch: {
    page() {
      this.updateList();
    },
  },
  mounted() {
    this.updateList();
  },
  methods: {
    isSuccess(item) {
      return (item.intent_status || item.status) === 'success' && (!item.entity_status || item.entity_status === 'success');
    },
    ...mapActions([
      'getAllResultsLog',
    ]),
    async updateList() {
      this.busy = true;
      try {
        const response = await this.getAllResultsLog({
          repositoryUuid: this.repository.uuid,
          resultId: this.id,
          page: this.page,
        });
        const data = response.data.log;
        this.resultExampleList = data.results;
        this.pages = data.total_pages;
        this.error = null;
      } catch (error) {
        this.error = error;
      } finally {
        this.busy = false;
      }
    },
  },
};
</script>

<style lang="scss" scoped>
.evaluate-result-example-list {
  width: 100%;

  &__title {
    margin-top: 1rem;
    font-weight: 700;
  }

  &__loading {
    width: 100%;
    text-align: center;
  }

  &__pagination {
    min-width: 100%;
    display: block;
    margin: 1rem auto;
    max-width: 600px;
    display: flex;
    justify-content: flex-end;
  }
}
.no-examples {
  margin: 8px;
}
</style>
