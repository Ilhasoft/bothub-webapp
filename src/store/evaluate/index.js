import actions from './actions';
import state from './state';
import mutations from './mutations';
import getters from './getters';
import types from './types';

export default {
  actions,
  state,
  mutations,
  getters,
  types,
};
