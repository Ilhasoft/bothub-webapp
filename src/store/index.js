import Vue from 'vue';
import Vuex from 'vuex';

import Auth from './auth';
import User from './user';
import Org from './org';
import Repository from './repository';
import Category from './category';
import Example from './example';
import EvaluateExample from './evaluate-example';
import Translate from './translate';
import CachedFetch from './cached-fetch';
import Entity from './entity';
import Intent from './intent';
import News from './news';
import Tutorial from './tutorial';
import External from './external';

Vue.use(Vuex);

const store = new Vuex.Store({
  modules: {
    Auth,
    User,
    Org,
    Repository,
    Category,
    Example,
    Translate,
    CachedFetch,
    EvaluateExample,
    Entity,
    News,
    Tutorial,
    External,
    Intent,
  },
});

store.dispatch('retriveAuthToken');

export default store;
