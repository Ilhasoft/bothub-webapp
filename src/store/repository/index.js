import state from './state';
import getters from './getters';
import actions from './actions';
import mutations from './mutations';
import types from './types';

export default {
  state,
  getters,
  actions,
  mutations,
  types,
};
