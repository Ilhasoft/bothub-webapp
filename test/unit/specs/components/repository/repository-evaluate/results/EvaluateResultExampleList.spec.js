import { shallowMount, createLocalVue } from '@vue/test-utils';
import EvaluateResultExampleList from '@/components/repository/repository-evaluate/results/EvaluateResultExampleList';
import store from '@/store';


const localVue = createLocalVue();

describe('EvaluateResultExampleList.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(EvaluateResultExampleList, {
      localVue,
      propsData: {
        repository: {
          uuid: '8511fd26-a3bc-4f74-9af1-176abca5401d',
        },
        id: 2,
      },
      store,
    });
  });

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
