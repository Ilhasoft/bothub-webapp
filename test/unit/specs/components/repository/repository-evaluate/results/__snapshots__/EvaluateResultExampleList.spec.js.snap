// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EvaluateResultExampleList.vue renders correctly 1`] = `
<div class="evaluate-result-example-list">
  <h3 class="evaluate-result-example-list__title">
    some specific text
  </h3>
  <p>
    some specific text
  </p>
  <div class="evaluate-result-example-list__loading">
    <loading-stub></loading-stub>
  </div>
  <div class="evaluate-result-example-list__pagination">
    <b-pagination-stub total="0" perpage="20" current="1" rangebefore="1" rangeafter="1" iconprev="chevron-left" iconnext="chevron-right" arianextlabel="Next page" ariapreviouslabel="Previous page" ariapagelabel="Page" ariacurrentlabel="Current page"></b-pagination-stub>
  </div>
</div>
`;
